### PR TITLE
automatic token fetching

### PR DESCRIPTION
**Pull request by bshaffer**
see #809

### CODE REVIEW

File: tests/BaseTest.php [hunk 1]
```diff
@@ -110,19 +112,41 @@ public function tryToGetAnAccessToken(Google_Client $client)
...
 
-    $client->setRedirectUri("urn:ietf:wg:oauth:2.0:oob");
-    $client->setConfig('access_type', 'offline');
     $authUrl = $client->createAuthUrl();
 
-    echo "\nPlease enter the auth code:\n";
-    ob_flush();
+    $log = tempnam(sys_get_temp_dir(), 'php_auth_log');
+    $cmd = sprintf(
+      'OAUTH_LOG_FILE=%s php -S localhost:8000 -t %s > /dev/null & echo $!',
```
Comment (tmatsuo) [inline tests/BaseTest.php]: Does this command work on Windows?

Comment (tmatsuo) [inline tests/BaseTest.php]: I'm concerning the envvar

File: tests/BaseTest.php [hunk 2]
```diff
@@ -110,19 +112,41 @@ public function tryToGetAnAccessToken(Google_Client $client)
...
+    if ('1' === $file) {
+      $this->markTestSkipped('Failed to retrieve access token');
+    }
+
+    if (!$accessToken = json_decode($file, true)) {
+      $this->markTestSkipped('invalid JSON: ' . $file);
+    }
+
+    // kill the web server
+    `kill $pid`;
```
Comment (tmatsuo) [inline tests/BaseTest.php]: If you hit Ctrl+C while waiting for the token, what happens with the webserver process?

File: tests/BaseTest.php (hunk 1 above)
Comment (bshaffer) [inline tests/BaseTest.php]: it almost certainly does not work on windows, although I haven't checked.

**Comment by bshaffer** [comment]
Yeah I don't think we need this.

**Pull request closed without merging**